xbmc/windowing/gbm/drm/DRMUtils.cpp

```diff
@@ -251,16 +260,24 @@ bool CDRMUtils::InitGuiPlane(CEGLContextUtils* eglContext, EGLint renderableType
   m_video_plane = nullptr;
   m_crtc = nullptr;
 
+  struct
+  {
+    uint32_t drmformat;
+    std::vector<uint32_t> eglformats;
+    bool alpha;
+  } typedef formatmap;
+
   // add more formats here in future
-  std::map<std::uint32_t, std::vector<uint32_t>> formats{
-      {DRM_FORMAT_ARGB8888, {DRM_FORMAT_ARGB8888, DRM_FORMAT_XRGB8888}},
+  std::vector<formatmap> formats{
+      {DRM_FORMAT_ARGB8888, {DRM_FORMAT_ARGB8888, DRM_FORMAT_XRGB8888}, true},
+      {DRM_FORMAT_XRGB8888, {DRM_FORMAT_ARGB8888, DRM_FORMAT_XRGB8888}, false},
```

Comment:
mesa always uses ARGB/XRGB plane order, are there any funky EGL implementation that swaps the color order  like ABGR or BGRX and so on? Such DRM plane formats already exist but i am not so sure we will face any EGL implementation that provides such visual ids? If there are such combinations, they can be added here..